modules/llm-cache/ds/kv_state_cache.cc

```diff
@@ -110,16 +117,24 @@ KVStateCacheBuilder::KVStateCacheBuilder(Client& client,
     treeData->kvStateCacheBlockBuilder = kvStateCacheBlockBuilder;
     treeData->isPtr = true;
   }
+
+  kvStateCacheBuilder =
+      std::shared_ptr<KVStateCacheBuilder>(new KVStateCacheBuilder(
+          client, cache->GetDimension(), cache->GetLayer(), cache->rootTree));
```

Comment:
No `std::shared_ptr<T>(new T(...))`, use `std::make_shared<T>(...)`.

Please double check all such occurences in llm-cache/ module.

Reply:
OK